### PR TITLE
Remove SOMADataFrame; rename SOMAIndexedDataFrame to SOMADataframe

### PR DESCRIPTION
**Pull request by johnkerl**
As we discussed. See also https://github.com/single-cell-data/TileDB-SOMA/pull/537 and https://github.com/single-cell-data/TileDB-SOMA/pull/538.

### CODE REVIEW

File: abstract_specification.md [hunk 1]
```diff
@@ -250,117 +238,30 @@ Parameters:
...
 > ⚠️ **To be further specified** -- all methods need specification.
 >
 > Summary of operations:
 
 | Operation                               | Description                                                                    |
 | --------------------------------------- | ------------------------------------------------------------------------------ |
-| create(uri, ...)                        | Create a SOMAIndexedDataFrame.                                                 |
-| delete(uri)                             | Delete the SOMAIndexedDataFrame specified with the URI.                        |
-| exists(uri) -> bool                     | Return true if object exists and is a SOMAIndexedDataFrame.                    |
+| create(uri, ...)                        | Create a SOMADataFrame       .                                                 |
```
Comment (thetorpedodog) [inline abstract_specification.md:247]: wheeeee
```suggestion
| create(uri, ...)                        | Create a SOMADataFrame.                                                        |
```

**Review by thetorpedodog — APPROVED** [review]

File: abstract_specification.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ The data model is comprised of two layers:
 The foundational types are:
 
 - SOMACollection - a string-keyed container (key-value map) of other SOMA data types, e.g., SOMADataFrame, SOMADataMatrix and SOMACollection.
-- SOMADataFrame and SOMAIndexedDataFrame - a multi-column table -- essentially a dataframe, available in types which support offset indexing/slicing, or indexing based upon user-defined columns.
+- SOMADataFrame - a multi-column table -- essentially a dataframe, available in types which support offset indexing/slicing, or indexing based upon user-defined columns.
```
Comment (bkmartinjr) [inline abstract_specification.md:31]: This is incorrect - with the deletion of the dense dataframe, we no longer offer offset indexing/slicing.

Suggested rewording:   SOMADataFrame - a multi-column table -- essentially a dataframe with indexing on user-specified columns.

File: abstract_specification.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ In this document, the term `dataframe` implies something akin to an Arrow `Table
 - multiple columns may exist, each with a string column name
 - all columns are individually typed and contain simple data types (e.g., int64)
 - all columns are of equal length
-- rows are addressed by offset (SOMADataFrame), or by one or more dataframe columns (SOMAIndexedDataFrame)
+- rows are addressed by one or more dataframe columns (SOMADataFrame)
```
Comment (bkmartinjr) [inline abstract_specification.md:43]: You can remove the parenthetical at the end of the sentence, as it is no longer required to disambiguate the clauses...

File: abstract_specification.md [hunk 4]
```diff
@@ -250,117 +238,30 @@ Parameters:
...
-| exists(uri) -> bool                     | Return true if object exists and is a SOMAIndexedDataFrame.                    |
+| create(uri, ...)                        | Create a SOMADataFrame.                                                        |
+| delete(uri)                             | Delete the SOMADataFrame specified with the URI.                               |
+| exists(uri) -> bool                     | Return true if object exists and is a SOMADataFrame.                           |
 | get metadata                            | Access the metadata as a mutable [`SOMAMetadataMapping`](#SOMAMetadataMapping) |
-| get soma_type                           | Returns the constant "SOMAIndexedDataFrame"                                    |
+| get soma_type                           | Returns the constant "SOMADataFrame"                                           |
 | get schema -> Arrow.Schema              | Return data schema, in the form of an Arrow Schema                             |
 | get is_indexed -> bool                  | Return true.                                                                   |
```
Comment (bkmartinjr) [inline abstract_specification.md:253]: we don't need this method anymore - there is never a case where it is distinguishing...

**Review by bkmartinjr — CHANGES_REQUESTED** [review]
A few minor points where obsolete text was missed.  Otherwise it looks fine - thank you for doing this!

File: abstract_specification.md (hunk 4 above)
Comment (johnkerl) [reply abstract_specification.md:253]: @bkmartinjr we do need it (it's for all classe); I think you're thinking of `is_indexed`

Comment (johnkerl) [reply abstract_specification.md]: ... which I just removed on my last commit -- thanks!

**Comment by johnkerl** [comment]
@bkmartinjr ready for next round :)

File: abstract_specification.md (hunk 4 above)
Comment (bkmartinjr) [reply abstract_specification.md]: Yes, I made the comment on the `is_indexed` line - you guessed right.  Sorry for being opaque!

**Review by bkmartinjr — APPROVED** [review]
LGTM.  Thank you!

**Pull request merged**